src/lando/headless_api/tests/test_automation_api.py

```diff
@@ -885,6 +885,256 @@ def test_automation_job_merge_onto_fail(
+    request,
+    monkeypatch,
+    normal_patch,
+):
+    repo = repo_mc(SCM_TYPE_GIT)
+
+    # Create a new commit that will be tagged
+    _create_git_commit(request, Path(repo.system_path))
+
+    tag_name = "v-tagtest-git"
+
+    job = AutomationJob.objects.create(
+        status=JobStatus.SUBMITTED,
+        requester_email="test@example.com",
+        target_repo=repo,
+    )
+    AutomationAction.objects.create(
+        job_id=job,
+        action_type="add-commit",
+        data={
+            "action": "add-commit",
+            "content": PATCH_GIT_1,
+            "patch_format": "git-format-patch",
+        },
+        order=0,
+    )
+    AutomationAction.objects.create(
+        job_id=job,
+        action_type="tag",
+        data={"action": "tag", "name": tag_name, "target": None},
+        order=1,
+    )
+
+    automation_worker = get_automation_worker(SCM_TYPE_GIT)
+    automation_worker.worker_instance.applicable_repos.add(repo)
+
+    assert automation_worker.run_automation_job(job)
+    assert job.status == JobStatus.LANDED
+
+    # Tag should be on the most recent commit.
+    expected_commit = subprocess.run(
+        ["git", "rev-parse", "HEAD"],
+        cwd=repo.system_path,
+        capture_output=True,
+        check=True,
+        text=True,
+    ).stdout.strip()
+
+    # now compare to the tag
+    tag_commit = subprocess.run(
+        ["git", "rev-list", "-n", "1", tag_name],
+        cwd=repo.system_path,
+        capture_output=True,
+        check=True,
+        text=True,
+    ).stdout.strip()
+
+    assert tag_commit == expected_commit
+
+
+@pytest.mark.django_db
+def test_automation_job_tag_failure_git(
+    repo_mc,
+    treestatusdouble,
+    get_automation_worker,
+    request,
+    monkeypatch,
+    normal_patch,
+):
+    repo = repo_mc(SCM_TYPE_GIT)
+
+    # Create a new commit that will be tagged
+    _create_git_commit(request, Path(repo.system_path))
+
+    tag_name = "v-tagtest-git"
+
+    job = AutomationJob.objects.create(
+        status=JobStatus.SUBMITTED,
+        requester_email="test@example.com",
+        target_repo=repo,
+    )
+    AutomationAction.objects.create(
+        job_id=job,
+        action_type="add-commit",
+        data={
+            "action": "add-commit",
+            "content": PATCH_GIT_1,
+            "patch_format": "git-format-patch",
+        },
+        order=0,
+    )
+    AutomationAction.objects.create(
+        job_id=job,
+        action_type="tag",
+        data={"action": "tag", "name": tag_name, "target": "bad-target"},
+        order=1,
+    )
+
+    automation_worker = get_automation_worker(SCM_TYPE_GIT)
+    automation_worker.worker_instance.applicable_repos.add(repo)
+
+    assert not automation_worker.run_automation_job(job)
+    job.refresh_from_db()
+    assert job.status == JobStatus.FAILED
+    assert "Aborting, could not perform `tag`, action #1" in job.error
+
+
+@pytest.mark.django_db
+def test_automation_job_tag_success_hg(
```

Comment:
This test is very similar to `test_automation_job_tag_success_git_new_commit`, and could benefit from being parameterized or use of fixtures to avoid repetition.